src/main/host/descriptor/socket/unix.rs

```diff
@@ -1320,6 +1444,17 @@ impl UnixSocketCommon {
     }
 }
 
+fn backlog_to_queue_size(backlog: i32) -> u32 {
+    // linux also makes this cast, so negative backlogs wrap around to large positive backlogs
+    // https://elixir.free-electrons.com/linux/v5.11.22/source/net/unix/af_unix.c#L628
```

Comment:
Is this a bug or a feature in Linux? If it's a bug, maybe worth logging a warning when backlog is negative?

Reply:
I don't think it's a bug or a feature, just a legacy artifact that probably can't be changed without breaking some userland programs. I don't think a warning is needed as it's not a bug or issue in shadow.